src/Pim/Component/Catalog/Model/AbstractProductValue.php

```diff
@@ -582,7 +582,7 @@ public function getPrices()
 
         ksort($prices);
 
-        return new ArrayCollection($prices);
+        return new PriceCollection($prices);
```

Comment:
I think you can simply return the collection
but check everywhere getPrices is used